### ISSUE TITLE
What does the option "Normalize variables"  of the PCA Widget does exactly ?

### ISSUE THREAD

**Issue by IKetchup** [biolab/orange3#5482]
- [ ] **What's wrong?**

Hello guys im having trouble to understand how to reproduce the option "Normalize variables" of the PCA widget in a python script with the PCA from SKlearn. How do I reproduce this option in a python script ?
I have try to normalise my dataframe with ```normalized_df=(df-df.mean())/df.std()``` but it doesn't return the same data.

I also tried to make my normalisation columnwise (advise of markotoplak on discord) as follow:

```
for col in df2.columns:
    df2[col] = (df2[col]-df2[col].mean())/df2[col].std()
```
In orange when I select this option on the widget PCA I have this:

![pb1](https://user-images.githubusercontent.com/72007646/121720325-5fe28600-cae3-11eb-8399-55bb23d3dcc1.png)

But in my python script with normalisation and PCA I have this:

![pb2](https://user-images.githubusercontent.com/72007646/121720588-94564200-cae3-11eb-9da1-37fe4c44dafd.png)

Does someone knows what the option "Normalize variables" of the PCA widget actually do ?
This information could help me reproduce the results I have in Orange (which is my primary goal).
- [ ] **What's your environment?**

- Operating system: Windows 10
- Orange version: 3.28.0
- How you installed Orange: Official website

**Comment by markotoplak** [comment]
Thanks! So I think that it is just divided by the standard deviation (so that we can still support sparse data). I tried it manually with this code:

```import Orange
from Orange.projection import PCA
from Orange.preprocess import Normalize

data = Orange.data.Table("iris")
data = Orange.data.Table.concatenate([data[:5], data[-5:]])

print("data")
print(data)

pca = PCA(n_components=2)
print("no normalizaiton")
pcam = pca(data)(data)
print(pcam)

pca = PCA(n_components=2,
          preprocessors=[Normalize(center=False)])
print("normalized Orange way")
pcam = pca(data)(data)
print(pcam)

data.X /= data.X.std(axis=0)
pca = PCA(n_components=2)
print("manual normalizaiton with std + Orange PCA")
pcam = pca(data)(data)
print(pcam)

print("use sklearn directly")
import sklearn.decomposition
pcas = sklearn.decomposition.PCA(n_components=2)
pcask = pcas.fit_transform(data.X)
print(pcask)
```

Any way of normalizing gets me the same results.

But I do agree the type of normalization needs to be documented, so please keep this bug open if you solve your immediate problems.

**Comment by IKetchup** [comment]
I have tried to divide by the standard diviation but it doesn't work. Why the Orange PCA using ```pca = PCA(n_components=2,
          preprocessors=[Normalize(center=False), Orange.preprocess.Continuize(),
                     Orange.preprocess.SklImpute()])````I get different result but still not the same as the ones I got in Orange

**Comment by IKetchup** [comment]
@markotoplak did you find something that could help me reproduce my results ? If you need anything like my data or my python file I will be glad to receive your help.

**Comment by markotoplak** [comment]
@IKetchup, you should get the same results if you used the same preprocessor order as in Orange. It is Continuize, SklImpute and then Normalize.
